src/lib/sync-request/worker.js

```diff
@@ -0,0 +1,39 @@
+const { request, ProxyAgent, setGlobalDispatcher } = require('undici')
+
+if (process.env.http_proxy || process.env.https_proxy) {
+  const proxyAgent = new ProxyAgent(process.env.http_proxy || process.env.https_proxy)
+  setGlobalDispatcher(proxyAgent)
+}
+
+function init() {
+  return async (options) => {
+    try {
+      const requestPromise = request(options.url, {
+        method: options.method || 'GET',
+        headers: options.headers,
+        body: options.body,
+      })
+      const timeoutPromise = new Promise((resolve, reject) => {
+        setTimeout(() => {
+          reject(new Error('Timeout'))
+        }, options.timeout * 1000)
+      })
```

Comment:
Undici supports `headersTimeout` and `bodyTimeout`:
https://undici.nodejs.org/#/docs/api/Dispatcher?id=parameter-dispatchoptions

Reply:
Neither of these parameters is a real timeout.